mongo/pagination/pagination_by_cursor.go

```diff
@@ -0,0 +1,103 @@
+package pagination
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Scalingo/go-utils/mongo/document"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+func NewByCursorService(opts ServiceOpts) ServiceByCursor {
+	return opts
+}
+
+type PaginateByCursorOpts struct {
+	Cursor      bson.M
+	AmountItems int
+	Query       bson.M
+	SortOrder   string
+}
+
+type ServiceByCursor interface {
+	PaginateByCursor(ctx context.Context, collection string, result interface{}, opts PaginateByCursorOpts) error
+}
+
+func (s ServiceOpts) paramValidationByCursor(collection string, opts *PaginateByCursorOpts) error {
+	badRequestErr := NewBadRequestErrors()
+	perPageErr := "per_page"
+
+	// Options validation
+	if s.PerPageDefault > s.MaxPerPage ||
+		s.PerPageDefault <= 0 ||
+		s.MaxPerPage <= 0 {
+		return errors.New("invalid pagination service configuration: MaxPerPage > PerPageDefault > 0")
+	}
+	// Parameter validation
+	if collection == "" {
+		return errors.New("invalid pagination service configuration: collection must be set")
+	}
+
+	// Default values assignation
+	if opts.AmountItems == 0 {
+		opts.AmountItems = s.PerPageDefault
+	}
+	if opts.SortOrder == "" {
+		opts.SortOrder = "-_id"
+	}
```

Comment:
I'm sorry I only notice this now. But I think that defining some default values should not happen in the `paramValidationByCursor`. When I read validation, I expect the method to check things, not modify things.